projects/filigran-website/components/icon-popup.tsx

```diff
@@ -0,0 +1,72 @@
+import React, { FC, ReactElement, useState } from 'react';
+
+interface PopupProps {
+    icon: ReactElement | null;
+    iconName: string;
+    onClose: () => void;
+}
+
+const Popup: FC<PopupProps> = ({ icon, iconName, onClose }) => {
+    const [copySuccess, setCopySuccess] = useState(false);
+
+    const handleCopy = () => {
+        const statement = `import { ${iconName} } from '@filigran-ui';`;
+        navigator.clipboard.writeText(statement)
+            .then(() => {
+                setCopySuccess(true);
+                setTimeout(() => setCopySuccess(false), 2000);
+            })
+            .catch(err => console.error('Failed to copy:', err));
+    };
+
+    if (!icon) return null;
+
+    const handleBackgroundClick = (e: React.MouseEvent<HTMLDivElement>) => {
+        if (e.target === e.currentTarget) {
+            onClose();
+        }
+    };
+
+    return (
+        <div className="fixed inset-0 z-50 flex items-center justify-center p-4 bg-black bg-opacity-50" onClick={handleBackgroundClick}>
```

Comment:
You can use instead.
https://filigranhq.github.io/filigran-ui/docs/components/dialog 
I would prefer to use our own component of the ui-library to make sense.

Docs :https://ui.shadcn.com/docs/components/dialog